wireapp/wire-android
# fix: Calling video not streamed when enabling camera on preview screen (WPB-7114)

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-7114" title="WPB-7114" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-7114</a>  [Android] When video is enabled on precall screen, video is not streamed during 1:1 and group call
  </td></table>
  <br />
 
























----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

Calling video not streamed when enabling camera on preview screen of an incoming call.

### Causes (Optional)

TLDR: A race condition.
Establishing an incoming calls takes some times an important time, causing to call the function of video stream before the call get established.

### Solutions

I moved the logic of sending video feed to OngoingCallViewModel, this way we are sure `setVideoSendState` will be called only when the call get established.

Needs releases with:


### Testing

#### Test Coverage (Optional)


#### How to Test

- Start a call to an android device.
- On Android enable your camera and accept the call
- You should still see your camera on, and the other person see your video feed.

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
